src/main/org/bson/ByteBuf.java

```diff
@@ -98,6 +98,11 @@
    */
   private int writePosition = 0;
 
+  /**
+   * The size of an unsigned 32-bit integer: 2^32
+   */
+  private static long UINT32_SIZE = (long)Math.pow(2, 32);
```

Comment:
Can you please calculate the 2^32-1 and provide it here as a literal (not 2^32), and rename the variable to UINT32_MAX to match the C code.